src/lib/json-ld.js

```diff
@@ -0,0 +1,107 @@
+import { Helmet } from 'react-helmet';
+
+import config from '../../package.json';
+
+export function ArticleJsonLd({ post = {}, siteTitle = '' }) {
+  const { homepage = '', favicon = 'favicon.ico' } = config;
+  const { title, slug, excerpt, date, author, categories, modified, featuredImage } = post;
+
+  /** TODO - As image is a recommended field would be interesting to have a
+   * default image in case there is no featuredImage comming from WP,
+   * like the open graph social image
+   * */
+
+  const jsonLd = {
+    '@context': 'https://schema.org',
+    '@type': 'Article',
+    mainEntityOfPage: {
+      '@type': 'WebPage',
+      '@id': `${homepage}/posts/${slug}`,
+    },
+    headline: title,
+    image: [featuredImage?.sourceUrl],
+    datePublished: new Date(date).toISOString(),
+    dateModified: new Date(modified).toISOString() || new Date(date).toISOString(),
```

Comment:
would the `||` statement actually work here? wouldn't it throw an error on the `Date` instance?

![image](https://user-images.githubusercontent.com/1045274/98877988-d1742c00-244f-11eb-85ba-9b06e43be671.png)

Reply:
Oops, didnt think on that. I think a good idea would be to lift date validation at the top, also because now checking with fresh eyes I'm seeing that new Date(date) is being potentially created 3 three times, instead of creating once and reusing it. Will make the changes 👍